sdk/export/metric/metric.go

```diff
@@ -171,6 +171,82 @@ type Exporter interface {
 	Export(context.Context, CheckpointSet) error
 }
 
+// Convenience function that creates a slice of labels from the passed
+// iterator. The iterator is set up to start from the beginning before
+// creating the slice.
+func IteratorToSlice(iter LabelIterator) []core.KeyValue {
+	l := iter.Len()
+	if l == 0 {
+		return nil
+	}
+	iter.idx = -1
+	slice := make([]core.KeyValue, 0, l)
+	for iter.Next() {
+		slice = append(slice, iter.Label())
+	}
+	return slice
+}
+
+type LabelStorage interface {
+	NumLabels() int
+	GetLabel(int) core.KeyValue
+}
+
+type LabelSlice []core.KeyValue
+
+var _ LabelStorage = LabelSlice{}
+
+func (s LabelSlice) NumLabels() int {
+	return len(s)
+}
+
+func (s LabelSlice) GetLabel(idx int) core.KeyValue {
+	return s[idx]
+}
+
+// LabelIterator allows iterating over ordered set of labels. The
+// typical use of the iterator is as follows:
+//
+//     iter := export.NewLabelIterator(getStorage())
+//     for iter.Next() {
+//       label := iter.Label()
+//       // or, if we need an index:
+//       // idx, label := iter.IndexedLabel()
+//       // do something with label
+//     }
+type LabelIterator struct {
+	storage LabelStorage
+	idx     int
+}
+
+func NewLabelIterator(storage LabelStorage) LabelIterator {
+	return LabelIterator{
+		storage: storage,
+		idx:     -1,
+	}
+}
+
+func (i *LabelIterator) Next() bool {
+	i.idx++
+	return i.idx < i.Len()
+}
+
+func (i *LabelIterator) Label() core.KeyValue {
+	return i.storage.GetLabel(i.idx)
+}
+
+func (i *LabelIterator) IndexedLabel() (int, core.KeyValue) {
+	return i.idx, i.Label()
+}
+
+func (i *LabelIterator) Len() int {
+	return i.storage.NumLabels()
+}
+
+func NewSliceLabelIterator(s []core.KeyValue) LabelIterator {
```

Comment:
Could we shorten this name? Is this likely to be used in non-test code?  Another idea might be to define this as a method with a `LabelSlice` receiver.  Callers could write `metric.LabelSlice(xyz).NewIterator()`.

Reply:
Yes, it's quite a mouthful. Not sure about using it in non-test code, maybe some other SDK implementation will use it?

Good idea with the method. Will use `Iter` name - shorter and same as name as in `export.Labels`.